### DISCUSSION TITLE
Zwave signal strength

### DISCUSSION THREAD

**Discussion by iptvcld** [tsightler/ring-mqtt#443] [General Discussion]
Hello, is it possible to see the zwave signal for door/window sensors as well as keypads gen2?

**Comment by tsightler** [comment]
There is commStatus and linkQuality values returned as part of the info sensor, but it's not a signal level, just OK or not.  That's all that seems to be available in the API.

**Reply by iptvcld** [reply]
Thank you, is it possible to see what sensor is connected to either the base station or the range Extender? When I call unto ring they have all this helpful information.

**Reply by tsightler** [reply]
Possibly, there's some data available in the API around this, for example, the Z-wave adapter device has a property ("network") which appears to be a JSON object describing the Z-wave nodes, and the extender may have something similar, so I'd guess you could parse that an probably map some hierarchy, but I have no idea for sure as I've not dug into it that deeply.  You are welcome to explore the data yourself by using the ring-client-api project, even just running the device discovery from that project would give you an idea of what data is available.

**Reply by iptvcld** [reply]
Sorry.. When you say Z-wave adapter - is that the Ring Base Station?  You mentioned it has a property ("network") but I am not able to locate that on the base station. Thanks

**Reply by tsightler** [reply]
The base station actually has both a Z-wave and Zigbee adapter built in, and those do show up as their own devices in the API.  Look for a device with the type: 'adapter.zwave'.

**Reply by iptvcld** [reply]
Sorry; I am looking at the Ring-MQTT integration in HA.  I am going to Integrations and I click on MQTT, thats where it shows me all the sensors discovered.  But I am not able to locate 'adapter.zwave' in this section.  Should I be looking somewhere else?

**Reply by tsightler** [reply]
There is nothing available in ring-mqtt around this.  As I stated above, the data I'm referring to is available via the Ring API and you can use ring-client-api to explore it if you like, or just run the data discovery tool from the ring-client-api project which will dump out the data available.

**Reply by iptvcld** [reply]
ahh ok, got it - I will set that up and see what it dumps out. Thank you

**Comment by iptvcld** [comment]
Appreciate the reply and help. Amazing adding this is BTW..
